File: backend/core-service/src/Fix.Application/UseCases/Policies/Commands/AddPolicyInstrumentCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Policies.Dtos;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Policies.Commands;

[RequireRole(RoleCodes.UpdatePolicy)]
public sealed record AddPolicyInstrumentCommand(
    Guid UserId,
    Guid OrganizationId,
    Guid PolicyId,
    PolicyInstrumentInput Instrument)
    : ICommand<PolicyDto>, IOrganizationRequest;

File: frontend/bff/src/application/policies/commands/add-policy-instrument.command.ts
import type { PolicyInstrumentInput } from './policy-instrument.input.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface AddPolicyInstrumentCommand {
  context: RequestContext;
  policyId: string;
  instrument: PolicyInstrumentInput;
}
